lms/djangoapps/grades/new/course_grade.py

```diff
@@ -24,8 +27,39 @@ class CourseGrade(object):
     def __init__(self, student, course, course_structure):
         self.student = student
         self.course = course
+        self.course_version = getattr(course, 'course_version', None)
+        self.course_edited_timestamp = getattr(course, 'subtree_edited_on', None)
         self.course_structure = course_structure
-        self.chapter_grades = []
+        self._percent = None
+        self._letter_grade = None
+        self.subsection_grade_factory = SubsectionGradeFactory(self.student, self.course, self.course_structure)
+
+    @classmethod
+    def load_persisted_grade(cls, user, course, course_structure, current_grading_policy_hash):
+        """
+        Initializes a CourseGrade object, filling its members with persisted values from the database.
+
+        If the grading policy is out of date, recomputes the grade.
+
+        If no persisted values are found, returns None.
+        """
+        try:
+            persistent_grade = PersistentCourseGrade.read_course_grade(user.id, course.id)
+        except PersistentCourseGrade.DoesNotExist:
+            return None
+        course_grade = CourseGrade(user, course, course_structure)
+
+        if current_grading_policy_hash != persistent_grade.grading_policy_hash:
```

Comment:
We do log calls to `compute_and_update`, the method we call if the hashes don't match. So I believe that we could infer pretty easily from our logs that this case was hit, by seeing a log for `compute_and_update` immediately preceding a log for `load_persisted_grade` for the same user, course, etc. That said, it would not be explicit. Do you think that is sufficient, or would an explicit indication within the log statement for `load_persisted_grade` be useful?